examples/axes_grid1/axes_margins.py

```diff
@@ -0,0 +1,31 @@
+"""
+============
+Margins demo
+============
+
```

Comment:
Can you please add a small paragraph explaining what the example does?
The title is also not very informative.

Reply:
Pushed https://github.com/matplotlib/matplotlib/pull/9459/commits/e3280c67845040bc1926d13d1160783a3d6ab74c. Is it enough or should I elaborate?

Reply:
Do you mind if I push some changes directly to the branch?

Reply:
Go ahead.